Patch-Source: https://gitweb.gentoo.org/repo/gentoo.git/plain/games-engines/openmw/files/openmw-0.48.0-gcc14.patch
The patch covers the first two hunks.
--
Fix build with gcc 14
https://bugs.gentoo.org/925105

--- a/components/bsa/bsa_file.cpp
+++ b/components/bsa/bsa_file.cpp
@@ -26,6 +26,7 @@
 #include <components/files/constrainedfilestream.hpp>
 
 #include <cassert>
+#include <algorithm>
 
 #include <boost/filesystem/path.hpp>
 #include <boost/filesystem/fstream.hpp>
--- a/apps/openmw/mwstate/charactermanager.hpp
+++ b/apps/openmw/mwstate/charactermanager.hpp
@@ -1,6 +1,8 @@
 #ifndef GAME_STATE_CHARACTERMANAGER_H
 #define GAME_STATE_CHARACTERMANAGER_H

+#include <list>
+
 #include <boost/filesystem/path.hpp>

 #include "character.hpp"
--- a/components/vfs/filesystemarchive.cpp
+++ b/components/vfs/filesystemarchive.cpp
@@ -1,5 +1,7 @@
 #include "filesystemarchive.hpp"
 
+#include <algorithm>
+
 #include <boost/filesystem.hpp>
 
 #include <components/debug/debuglog.hpp>
